apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Snippets/BigQueryDatasetSnippets.cs

```diff
@@ -0,0 +1,190 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Bigquery.v2.Data;
+using Google.Cloud.ClientTesting;
+using System;
+using System.Threading;
+using Xunit;
+
+namespace Google.Cloud.BigQuery.V2.Snippets
+{
+    [SnippetOutputCollector]
+    [Collection(nameof(BigQuerySnippetFixture))]
+    public class BigQueryDatasetSnippets
+    {
+        private readonly BigQuerySnippetFixture _fixture;
+
+        public BigQueryDatasetSnippets(BigQuerySnippetFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public void Update()
+        {
+            // Avoid quota issues: only one dataset modification is allowed per 2 seconds.
+            Thread.Sleep(2500);
+
+            string projectId = _fixture.ProjectId;
+            string datasetId = _fixture.GenerateDatasetId();
+
+            BigQueryClient.Create(projectId).CreateDataset(datasetId);
+
+            // Snippet: Update(Dataset, *)
+            BigQueryClient client = BigQueryClient.Create(projectId);
+            BigQueryDataset dataset = client.GetDataset(datasetId);
+
+            // This example modifies the in-memory resource in the BigQueryDataset,
+            // and then applies that change in the server. Alternatively, pass a Dataset
+            // into the Update method.
+            dataset.Resource.FriendlyName = "Updated dataset";
+            BigQueryDataset updated = dataset.Update();
+            
+            Console.WriteLine($"Updated dataset friendly name: {updated.Resource.FriendlyName}");
+            // End snippet
+
+            Assert.Equal("Updated dataset", updated.Resource.FriendlyName);
+        }
+
+        // See-also: Update(Dataset, *)
+        // Member: UpdateAsync(Dataset, *, *)
+        // See [Update](ref) for a synchronous example.
+        // End see-also
+
+        [Fact]
+        public void Patch()
+        {
+            // Avoid quota issues: only one dataset modification is allowed per 2 seconds.
+            Thread.Sleep(2500);
+
+            string projectId = _fixture.ProjectId;
+            string datasetId = _fixture.GenerateDatasetId();
+
+            BigQueryClient.Create(projectId).CreateDataset(datasetId);
+
+            // Snippet: Patch(Dataset, bool, *)
+            BigQueryClient client = BigQueryClient.Create(projectId);
+            BigQueryDataset dataset = client.GetDataset(datasetId);
+
+            // There's no ETag in this Dataset. The matchETag parameter in the method call
+            // determines whether the ETag in the original resource is propagated into the
+            // patch.
+            var patch = new Dataset
```

Comment:
I know it's obvious what the type is, but I'd rather there is no use of `var` in snippets.

Reply:
Agreed. These were copied and pasted, so I'll check the original source.